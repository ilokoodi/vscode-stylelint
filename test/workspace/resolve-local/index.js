'use strict';

const path = require('path');

const pWaitFor = require('p-wait-for');
const { workspace, window } = require('vscode');

const { normalizeDiagnostic, getStylelintDiagnostics } = require('../utils');

describe('Local Stylelint resolution', () => {
	it('should resolve to the locally installed copy of Stylelint', async () => {
		// Open the './test.css' file.
		const cssDocument = await workspace.openTextDocument(path.resolve(__dirname, 'test.css'));

		await window.showTextDocument(cssDocument);

		// Wait for diagnostics result.
		await pWaitFor(() => getStylelintDiagnostics(cssDocument.uri).length > 0, { timeout: 5000 });

		// Check the result.
		const diagnostics = getStylelintDiagnostics(cssDocument.uri);

		expect(diagnostics.map(normalizeDiagnostic)).toMatchSnapshot();
	});
});
