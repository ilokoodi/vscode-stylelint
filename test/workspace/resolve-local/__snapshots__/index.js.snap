// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local Stylelint resolution should resolve to the locally installed copy of Stylelint 1`] = `
Array [
  Object {
    "code": "fake",
    "data": undefined,
    "hasDiagnosticCode": false,
    "message": "Fake result",
    "range": Object {
      "end": Object {
        "character": 0,
        "line": 0,
      },
      "start": Object {
        "character": 0,
        "line": 0,
      },
    },
    "severity": 0,
    "source": "stylelint",
  },
]
`;
